### PR TITLE
docs, mkdocs: fix domain, title caps

### PR DESCRIPTION
**Pull request by woodruffw**
Fixes the sentence-casing of domains (and some small title inconsistencies) by adding an explicit `nav` hierarchy.

Downside: requires files to be tracked in the `nav` hierarchy as well before they get rendered. This downside is shared by the PyPI user + dev docs currently, and is a known limitation of mkdocs.

Before:

Screenshot 2024-07-02 at 4 49 28 PM

After:

Screenshot 2024-07-02 at 4 49 47 PM

CC @miketheman, who made me see these and activated my compulsion to fix them 🙂

### CODE REVIEW

**Comment by woodruffw** [comment]
(NB: I've also made an effort to retain the same `nav` order that `mkdocs` was already rendering the hierarchy with.)

File: docs/pypi.org/Terms-of-use.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 source: https://pypi.org/policy/terms-of-use/
 ---
-# Terms of use
+# Terms of Use
```
Comment (miketheman) [inline docs/pypi.org/Terms-of-use.md:4]: Note: if the case here changes, we'll need an update over here:
https://github.com/pypi/warehouse/blob/83b0c6de32c174208f3a65c061bd8fcc8983213c/warehouse/routes.py#L62 
(and associated test case) as mkdocs/rtd are case sensitive.

**Review by miketheman — APPROVED** [review]
I'm generally fine with this - as it does make it explicit and allows us to control the order.

@ewdurbin any objections?

File: docs/pypi.org/Terms-of-use.md
Comment (miketheman) [inline docs/pypi.org/Terms-of-use.md:1]: The filename needs to be updated for case sensitivity as well - currently this is a 404:
https://psf-policies--15.org.readthedocs.build/pypi.org/Terms-of-Use.md

Comment (woodruffw) [reply docs/pypi.org/Terms-of-use.md:1]: Ugh, I did this locally but it didn't commit thanks to macOS's case insensitivity.

File: docs/pypi.org/Terms-of-use.md (hunk 1 above)
Comment (woodruffw) [reply docs/pypi.org/Terms-of-use.md:4]: Cool, I'll make a PR there as well!

Comment (woodruffw) [reply docs/pypi.org/Terms-of-use.md:4]: https://github.com/pypi/warehouse/pull/16210

**Comment by ewdurbin** [comment]
maybe add a note to README "how to add a page to nav"?

**Comment by woodruffw** [comment]
> maybe add a note to README "how to add a page to nav"?

Done with 934e64e!

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**